qiskit/visualization/matplotlib.py

```diff
@@ -1372,52 +1368,32 @@ def _line(self, xy0, xy1, lc=None, ls=None, zorder=PORDER_LINE):
 class Anchor:
     """Locate the anchors for the gates"""
 
-    def __init__(self, reg_num, yind, fold):
-        self._yind = yind
+    def __init__(self, num_wires, y_index, fold):
+        self._num_wires = num_wires
         self._fold = fold
-        self._reg_num = reg_num
-        self._gate_placed = []
-        self.nxt_anchor_idx = 0
-        self.gate_anchor = 0
-
-    def plot_coord(self, index, gate_width, x_offset):
-        """Set the coord positions for an index"""
-        h_pos = index % self._fold + 1
+        self._y_index = y_index
+        self._x_index = 0
+
+    def plot_coord(self, x_index, gate_width, x_offset):
+        """Get the coord positions for an index"""
+        h_pos = x_index % self._fold + 1
         # check folding
         if self._fold > 0:
             if h_pos + (gate_width - 1) > self._fold:
-                index += self._fold - (h_pos - 1)
-            x_pos = index % self._fold + 0.5 * gate_width + 0.04
-            y_pos = self._yind - (index // self._fold) * (self._reg_num + 1)
+                x_index += self._fold - (h_pos - 1)
+            x_pos = x_index % self._fold + 0.5 * gate_width + 0.04
+            y_pos = self._y_index - (x_index // self._fold) * (self._num_wires + 1)
         else:
-            x_pos = index + 0.5 * gate_width + 0.04
-            y_pos = self._yind
+            x_pos = x_index + 0.5 * gate_width + 0.04
+            y_pos = self._y_index
 
         # could have been updated, so need to store
-        self.gate_anchor = index
+        self._x_index = x_index
         return x_pos + x_offset, y_pos
 
-    def set_index(self, index, layer_width):
-        """Set the index for a gate"""
-        if self._fold < 2:
-            _index = index
-        else:
-            h_pos = index % self._fold + 1
-            if h_pos + (layer_width - 1) > self._fold:
-                _index = index + self._fold - (h_pos - 1) + 1
-            else:
-                _index = index
-        for ii in range(layer_width):
-            idx = _index + ii
-            if idx not in self._gate_placed:
-                self._gate_placed.append(idx)
-                self.nxt_anchor_idx = idx + 1
-
-    def get_index(self):
-        """Getter for the index"""
-        if self._gate_placed:
-            return self._gate_placed[-1] + 1
-        return 0
+    def get_x_index(self):
+        """Getter for the x index"""
+        return self._x_index
```

Comment:
At this point it may as well be a property
```python
@property
def x_index(self):
    return self._x_index
```
but this isn't important.

Reply:
I left it as is unless there's a reason to add `@property`.

Reply:
It's more "Pythonic", but it really doesn't matter.